File: src/com/jepl/lang/libraries/io/io.java
package com.jepl.lang.libraries.io;

import com.jepl.lang.JEPLException;
import com.jepl.lang.Values;
import com.jepl.lang.libraries.Function;
import com.jepl.lang.libraries.Library;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.PrintWriter;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.HashMap;
import java.util.List;

public class io implements Library {
    @Override
    public HashMap<String, Function> invoke() {
        HashMap<String, Function> out = new HashMap<>();
        out.put("mkdir", args -> {
            final String dirname = args.get(0);
            File file = new File(dirname);
            if(!file.exists()){
                file.mkdir();
            }
        });
        out.put("delete", args -> {
            final String fname = args.get(0);
            File file = new File(fname);
            if(file.exists()) {
                file.delete();
            }else {
                throw new JEPLException(new RuntimeException("File "+fname+" isn't exists!"));
            }
        });
        out.put("io_put_contents", args -> {
            final String filename = args.get(0);
            final String content = args.get(1);
            PrintWriter writer;
            try {
                writer = new PrintWriter(filename);
            } catch (FileNotFoundException e) {
                throw new JEPLException(e);
            }
            writer.print(content);
            writer.close();
        });
        out.put("io_get_contents", args -> {
            final String filename = args.get(0);
            final String var = args.get(1);
            try {
                Values.addVar(var, Files.readString(Path.of(filename)));
            } catch (IOException e) {
                throw new JEPLException(e);
            }
        });
        out.put("io_file_rename", args -> {
           Path src = Path.of(args.get(0));
           final String newName = args.get(1);
            try {
                Files.move(src, src.resolveSibling(newName));
            } catch (IOException e) {
                throw new JEPLException(new RuntimeException(e));
            }
        });
        return out;
    }
}

File: src/com/jepl/lang/libraries/io/io.kt
package com.jepl.lang.libraries.io

import com.jepl.lang.JEPLException
import com.jepl.lang.Values
import com.jepl.lang.libraries.Function
import com.jepl.lang.libraries.Library
import java.io.File
import java.io.FileNotFoundException
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Path

class io : Library {
    override fun invoke(): HashMap<String, Function> {
        val out = HashMap<String, Function>()
        out["io_mkdir"] = object : Function {
            override fun invoke(args: List<String>) {
                val dirname = args[0]
                val file = File(dirname)
                if (!file.exists()) {
                    file.mkdir()
                }
            }
        }
        out["io_delete"] = object : Function {
            override fun invoke(args: List<String>) {
                val fname = args[0]
                val file = File(fname)
                if (file.exists()) {
                    file.delete()
                } else {
                    throw JEPLException(RuntimeException("File $fname isn't exists!"))
                }
            }
        }
        out["io_put_contents"] = object : Function {
            override fun invoke(args: List<String>) {
                val filename = args[0]
                val content = args[1]
                val writer: File
                try {
                    writer = File(filename)
                } catch (e: FileNotFoundException) {
                    throw JEPLException(e)
                }
                writer.writeText(content)
            }
        }
        out["io_get_contents"] = object : Function {
            override fun invoke(args: List<String>) {
                val filename = args[0]
                val `var` = args[1]
                try {
                    Values.addVar(`var`, File(filename).readText())
                } catch (e: IOException) {
                    throw JEPLException(e)
                }
            }
        }
        out["io_file_rename"] = object : Function {
            override fun invoke(args: List<String>) {
                val src = Path.of(args[0])
                val newName = args[1]
                try {
                    Files.move(src, src.resolveSibling(newName))
                } catch (e: IOException) {
                    throw JEPLException(RuntimeException(e))
                }
            }
        }
        return out
    }
}